### DISCUSSION TITLE
Locking block placement

### DISCUSSION THREAD

**Discussion by kathy-li-ho** [tinacms/tinacms#2408] [Help] [answered]
Hi! 

So all the content blocks in Tina can be re-arranged, which is great, but is there a way I can lock a block to one position? For example, a hero block that should always be up top, or some testimonial block that should always be last?

For my client, less flexibility is better.

**Comment by cjmcintyre** [comment, marked as answer]
If you are referring to something like this?

![image](https://user-images.githubusercontent.com/18439866/148621245-f33d1cda-c126-4852-8fa9-bb5b094dac37.png)

I believe this is possible as I've seen it in some of the old docs https://pre-beta.tina.io/docs/reference/toolkit/fields/blocks/ which I believe is based from https://github.com/tinacms/tina-starter-grande

**Edit:** It is possible.
![image](https://user-images.githubusercontent.com/18439866/148622405-9f932b79-1f70-49b5-9a58-e006e7c42397.png)

Now this may be dependant on which Tina code / version you have pulled... but somewhere in the schema file you'll find a pages object. 

```js
{ 
      label: "Pages",
      name: "pages",
      path: "content/pages",
      fields: [
        {
          type: "object",
          list: true,
          name: "blocks",
          label: "Sections",
          templates: [
            heroBlockSchema,
            featureBlockShema,
            testimonialBlockSchema,
          ],
        },
      ],
    },
}
```

You'll want to make it accept another object, with our heroBlockSchema
```js

{
      label: "Pages",
      name: "pages",
      path: "content/pages",
      fields: [
        {
          type: "object",
          list: true,
          name: "blocks2",
          label: "Hero",
          templates: [
            heroBlockSchema,
          ]
        },
        {
          type: "object",
          list: true,
          name: "blocks",
          label: "Sections",
          templates: [
            heroBlockSchema,
            featureBlockShema,
            testimonialBlockSchema,
          ],
        },
      ],
    },

```
And then you'll want to modify the block file and add another switch to check for that block

```tsx

export const Blocks = (props: Pages) => {
	return (
		<>
			{props.blocks2
				? props.blocks2.map(function (block, i) {
					switch (block.__typename) {
						case 'PagesBlocks2Hero':
							return (
								<React.Fragment key={i + block.__typename}>
									<Hero data={block} />
								</React.Fragment>
							);
						default:
							return null;
					}
				}) : null}
			{props.blocks
				? props.blocks.map(function (block, i) {
					switch (block.__typename) {
```

![image](https://user-images.githubusercontent.com/18439866/148622998-6e41e642-170a-428e-8f69-30944eceae7d.png)

And now we have the option to add our hero block. If you wanted to force it, i'd say just don't use a switch statement.

**Reply by cjmcintyre** [reply]
Additionally something along the lines of this

```js
  { 
      label: "Pages",
      name: "pages",
      path: "content/pages",
      fields: [
        {
          label: "Hero2",
          name: "heroBlock",
          type: "object",
          fields: [
            {
              label: "Author",
              name: "author",
              type: "string"
            },
            {
              label: "Role",
              name: "role",
              type: "string"
            },
            {
              label: "Quote",
              name: "quote",
              type: "string",
              ui: {
                component: "textarea"
              }
            }
          ]
        },
```

```js
export const Blocks = (props: Pages) => {
	return (
		<>
			{props.heroBlock
				? props.heroBlock.role : null}

			{props.blocks2
				? props.blocks2.map(function (block, i) {
					switch (block.__typename) {
						case 'PagesBlocks2Hero':
```
gave me probably what you are more looking for

![image](https://user-images.githubusercontent.com/18439866/148624095-beecab2f-c370-452f-a551-aef11f23695d.png)


![image](https://user-images.githubusercontent.com/18439866/148624077-7f16fb10-1925-4f86-a6b6-2a9e65501dee.png)

**Comment by kathy-li-ho** [comment]
oh wow! this works!

**Comment by jamespohalloran** [comment]
I think the approach of splitting things into 2 separate blocks fields is a great one here!

I'll just add another option, which is to to use a single blocks field, but sort the blocks in your render component

Something like:
```
export const Blocks = (props: Pages) => {
	return (
		<>
			{props.blocks
				? props.blocks.sort((a, b) => a.__typename == 'Hero' ? -1 : 1).map(function (block, i) {
					switch (block.__typename) {
						case 'PagesBlocksHero':
							return (
								<Hero /?
							);
						case 'PagesBlocksFeatures:
							return (
								<Features /?
							);
					}
				}) : null}
```

Here, the editor will be able to rearrange the blocks (including the Hero block), but it the Hero component will always render on the top of the page.

For a rundown on the blocks field, if you haven't already found it I'd recommend checking out the [Editing with blocks doc]( https://tina.io/docs/editing/blocks/).
The [Blocks component API reference]( https://tina.io/docs/reference/toolkit/fields/blocks/) doc is currently a bit dated for working with the latest API.

**Reply by kathy-li-ho** [reply]
Thanks for this!
